pkg/components/imguploader/gcsuploader.go

```diff
@@ -65,35 +57,34 @@ func (u *GCSUploader) Upload(ctx context.Context, imageDiskPath string) (string,
 	fileName += pngExt
 	key := path.Join(u.path, fileName)
 
-	var client *http.Client
-
+	var client *storage.Client
 	if u.keyFile != "" {
 		u.log.Debug("Opening key file ", u.keyFile)
 		data, err := ioutil.ReadFile(u.keyFile)
 		if err != nil {
 			return "", err
 		}
 
-		u.log.Debug("Creating JWT conf")
-		conf, err := google.JWTConfigFromJSON(data, tokenUrl)
+		u.log.Debug("Creating Google credentials from JSON")
+		creds, err := google.CredentialsFromJSON(ctx, data)
 		if err != nil {
 			return "", err
 		}
 
-		u.log.Debug("Creating HTTP client")
-		client = conf.Client(ctx)
+		u.log.Debug("Creating GCS client")
+		client, err = storage.NewClient(ctx, option.WithCredentials(creds))
+		if err != nil {
+			return "", err
+		}
 	} else {
-		u.log.Debug("Key file is empty, trying to use application default credentials")
-		client, err = google.DefaultClient(ctx)
+		u.log.Debug("Creating GCS client with default application credentials")
+		client, err = storage.NewClient(ctx)
 		if err != nil {
 			return "", err
 		}
 	}
 
-	err = u.uploadFile(client, imageDiskPath, key)
-	if err != nil {
-		return "", err
-	}
+	err = u.uploadFile(ctx, client, imageDiskPath, key)
```

Comment:
You need to keep the error check.
```suggestion
	err = u.uploadFile(ctx, client, imageDiskPath, key)
	if err != nil {
		return "", err
	}
```